### ISSUE TITLE
[BUG] emmet, autocomplete, props suggestion doesn't work in some .vue files

### ISSUE THREAD

**Issue by AlexSHoffman** [vuejs/vetur#827]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: Win 10
- Vetur version: 0.12.5
- VS Code version: 1.24.1

## Problem

In some .vue files emmet, autocomplete, props suggestion stops working (.vue files with not much code in them aren't affected)

The number of lines doesn't seem to be the reason it stops working (perhaps tag formatting is what's causing it? Doing `f1 => format document` isn't fixing the issue).
 
In the problematic files, if you type `div` or `v-layout` or any other tag manually it won't suggest to autocomplete it and after you hit enter, it moves the cursor to the next line without creating a tag.

I've reinstalled it several times (including the manual .vsix method) but it didn't help

**UPDATE**
---
The problem is caused by something in `<script>` section.

When I remove everything inside `return { }`, it starts working again
```

export default {
  name: "main",
  data() {
    return {
      ...
    }
  }
}
```
---

## VUE Language Server - output

```
Vetur initialized
[Error - 5:26:31 PM] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: Cannot read property 'parameters' of undefined
  Code: -32603 
[Error - 5:26:32 PM] Request textDocument/definition failed.
  Message: Request textDocument/definition failed with message: Cannot set property 'typeParameters' of undefined
  Code: -32603 
[Error - 5:26:34 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'parameters' of undefined
  Code: -32603 
[Error - 5:26:40 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'parameters' of undefined
  Code: -32603 
```

**Comment by AlexSHoffman** [comment]
**UPDATE:** I just realized that it works fine in new .vue files but it **stops** working in some files with lots of code in them

**Comment by AlexSHoffman** [comment]
**UPDATE:** I just realized that it works fine in new .vue files but it **stops** working in some files with lots of code in them. The number of lines doesn't seem to matter, perhaps tag formatting is the reason it stops working. Doing `f1 => format document` isn't fixing the issue

**Comment by sunyuu** [comment]
@AlexSHoffman Hi. i have the similar problems, but mine is it doesn't work in some project. I have no idea how this happened. Can you tell me how to see the vue language server outputs.

**Comment by AlexSHoffman** [comment]
@sunyuu to see the **vue language server** output you just need to:
- Open the VScode terminal. `Ctrl + ~`
- Go to Output tab
- Choose the **vue language server** in the drop down menu

**Comment by kindlyfire** [comment]
I get the same error. Code to reproduce: [gist](https://gist.github.com/kindlyfire/5c93de5c2c67fe7ecad0f325ec4dc013)

I commented the code to show the function causing the problem. When the watcher for `value` is commented out, everything works fine and emmet works. When it is uncommented, it causes the error and emmet/autocomplete doesn't work. I also noticed commenting out the line 21 (`this.rValue = v`) causes it not to error. Changing the name of the watcher `rValue` also makes the error go away.

Errors (I found two different ones):

```
[Error - 7:05:46 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'parameters' of undefined
  Code: -32603 
[Error - 7:12:26 PM] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: Cannot set property 'typeParameters' of undefined
  Code: -32603 
```

**Comment by boolka** [comment]
I got the same error.
Vetur - 0.16.2
![veturhovermessagestrace](https://user-images.githubusercontent.com/7414525/53630634-90a79580-3c21-11e9-843e-5e5a850d0b0e.png)
![veturhoverfail](https://user-images.githubusercontent.com/7414525/53630642-943b1c80-3c21-11e9-8262-13780a097984.png)
@kindlyfire recommendations works

**Comment by octref** [comment]
This seems to be solved in TS 3.3, which you can get by:

- `yarn add -D typescript@3.3` in workspace
-  `"vetur.useWorkspaceDependencies": true`

See #682 for details.

![image](https://user-images.githubusercontent.com/4033249/54790386-2c11a200-4bf3-11e9-8ff0-f926f04ad05d.png)

This issue will be completely gone after #1163.

**Comment by sunny8080** [comment]
Same problem happened with me, for some `vue` files emmet suggestions were working but for some files not. In my case it was occurring, because of `lang=""` which is included in `<template>` tag. 

Earlier:
![sol1](https://github.com/vuejs/vetur/assets/55615100/f839121e-c8cb-4ed3-8f42-00131ff2a2a7)


After removing `lang=""`  :
![sol2](https://github.com/vuejs/vetur/assets/55615100/1927ffd8-f905-4e1b-a99c-295e8d178029)
